### ISSUE TITLE
continuous integration on Windows platform

### ISSUE THREAD

**Issue by IndrajeetPatil** [easystats/easystats#35]
Is there any particular reason why we are not checking package builds on Windows CI services like `AppVeyor`?

Currently, we only have `Travis` builds for `linux` and `macOS`, but it will be good to have the same for Windows platform. 

If you agree, I can set up `yaml` files for repos.

**Comment by DominiqueMakowski** [comment]
We struggle to pass the current ones and you would like us to have even moar failures? 😁 
I don't have any objections to it

**Comment by IndrajeetPatil** [comment]
It's also for selfish reasons because I have never owned anything other than a Windows machine :D 

Currently, a lot of tests either straight up don't run on my local machine or fail. So I am concerned that this is either because development versions of some packages we are relying on loaded in my system or some legit other issues that specifically arise for a Windows platform and AppVeyor will help us diagnose that.

**Comment by DominiqueMakowski** [comment]
I am surprised! Me (and Daniel?) also use windows, and usually, all master branches pass locally (at least at *my* home)... but we might be unaware of some issues, so the more CI coverage we have the better!

**Comment by strengejacke** [comment]
In my experience, Travis is rather unstable. I also use Windows only, but I do local checks and use win-builder if necessary. But Travis is at least most times good enough to check on Linux and some OS X versions.

Since it regularly fails on OS X devel, but not on Linux devel, I still activate all platforms for pkg I maintain, because I don't care having passing checks on Travis in such case.

**Comment by strengejacke** [comment]
But honestly, mainly pkg should compile on a machine. Testing is in particular important before submitting stable versions to CRAN.

**Comment by IndrajeetPatil** [comment]
Some of the tests are indeed failing on Windows:
https://ci.appveyor.com/project/IndrajeetPatil/insight/build/job/28jhpgunliy5bf3g

**Comment by strengejacke** [comment]
Some of the models require some special attention, I can give it a try to fix some issues.

**Comment by strengejacke** [comment]
I'd say we remove Travis and rely on GH CI only, seems more reliable/stable

**Comment by DominiqueMakowski** [comment]
@IndrajeetPatil did you have any informative answers on your travis issue tweet?

**Comment by IndrajeetPatil** [comment]
I personally like Travis because I use it to detect lints and compute code coverage.
But if there is a way to do that using GitHub actions, I see no additional benefit to retaining Travis. 

I personally use both of them because Travis always has some issues and I need a second linux check, which is GitHub actions in my case. Another option would be Circle-CI, but there is not enough documentation to use it effectively. 

So, in summary, yeah, we can stop relying on Travis and fully switch to GitHub actions. :)

**Comment by DominiqueMakowski** [comment]
the main blocker is to get GHCI to trigger codecov as Indra said. Additionally, we could also program it to run styler automatically

**Comment by IndrajeetPatil** [comment]
There are some examples of `yml` files here which show how a lot of this can be implemented using actions:
https://github.com/r-lib/actions/tree/master/.github/workflows

**Comment by strengejacke** [comment]
I added the workflow to update the webpage after commit, seems to work great:
https://github.com/strengejacke/ggeffects/commit/c89bc2a31afb42e4fa71369b5068be9df043792d

🥂 

That's a nice workflow!

**Comment by strengejacke** [comment]
Ok, looks like is actually does _not_ work great 😠 
Would be nice if we get this workflow working.

**Comment by strengejacke** [comment]
So, now it looks as if it seems to work... This is a working pkgdown-workflow file: https://github.com/easystats/insight/blob/master/.github/workflows/pkgdown.yml

which builds the latest webpage on each push, committing it to the `gh-pages`-branch. In the settings, you just need to set this branch for the GitHub webpage (instead of `master/docs`).

**Comment by mattansb** [comment]
I have no idea what any of this means... 😅
So I'm thinking "copy `.github` folder from `insight` to `effectsize`"? Yes? No?

**Comment by DominiqueMakowski** [comment]
@mattansb it should work :)

**Comment by strengejacke** [comment]
We can now use "GitHub Actions", e.g. https://github.com/easystats/insight/actions

This is an automated set of workflows, like continuous integration (we used to use Travis for this). To use actions, you need to create a `.github`-folder with a subfolder `workflows` and create the yaml-files there, e.g. https://github.com/easystats/insight/blob/master/.github/workflows/R-check.yaml for automatic checks (like Travis CI).

Another workflow can use *pkgdown* to automatically build webpages, see https://easystats.github.io/insight/. If you create the above mentioned folder and copy [this file](https://github.com/easystats/insight/blob/master/.github/workflows/pkgdown.yml) into the folder, on each push to (master) GitHub, GitHub actions will automatically run `pkgdown::build_site()` to update your website, and pushes the files to the branch *gh-pages*. In the settings (https://github.com/easystats/insight/settings) under *GitHub Pages*, you select the `gh-pages` branch so you make sure to use that branch as source for the package-website.

**Comment by mattansb** [comment]
So no need to manually build the pkgdown site?? That's awesome!

**Comment by DominiqueMakowski** [comment]
Additionally, things that we could potentially add actions for, run styler to automatically fix syntax issues, and maybe automatically submit to CRAN win-builder (but for this kind o things it's better to do it less often, typically everytime dev is merged to master). Similarly, actions can also create automatic drafts for release including the changelog of all the issues closed (automated NEWS 🎉)

**Comment by mattansb** [comment]
> In the settings (https://github.com/easystats/insight/settings) under GitHub Pages, you select the gh-pages branch so you make sure to use that branch as source for the package-website.

I don't have access to the settings 🤔

**Comment by strengejacke** [comment]
yeah, rightly so! ;-)
I can make the changes...

**Comment by mattansb** [comment]
More than a year on the team, still no trust 🤣

**Comment by strengejacke** [comment]
I think we can close this. GH actions for checks works well, I'd say.
